### PR TITLE
Feature/architecture refactor phase 1

### PR DESCRIPTION
**Pull request by glckfndr**
What ### changed

- Simplized backend tags resolution pipeline.
- Split assistant responsibilities into focused modules:
- data/context loading
- scope resolution
- fallback resolution
- Kept AssistantService as orchestration-only flow.
- Added focused tests for scope and fallback behavior.

### Why

- Clarify ownership boundaries in assistant logic.
- Improve readability, testability, and future extensibility.
- Reduce service-level complexity and make behavior easier to reason about.

### Validation

- Ran full backend test suite: npm test (from backend/).
- Result: 18/18 test suites passed, 119/119 tests passed.

### Risks/Notes

- Main risk area is edge-case prompts around scope and fallback decisions.
- Mitigation: external behavior preserved; targeted tests added for scope/fallback decision paths.

### CODE REVIEW

File: docs/refactor-roadmap-feedback-2026-03-30.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+- classify -> build context -> call llm -> post-process
+
+3. Прибрати багаторівневий fallback pipeline, залишивши 1 прозору fallback-стратегію.
+4. Явно назвати межі відповідальності через типи/DTO між кроками.
+
+Валідація:
+
+- `cd backend && npm run test:run -- assistant.service`
+- `cd backend && npm run test:run -- assistant`
```
[the remark this answers is not in the document]
Comment (glckfndr) [reply docs/refactor-roadmap-feedback-2026-03-30.md]: Fixed.

File: docs/refactor-roadmap-feedback-2026-03-30.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+- backend/src/events/events-tags.helpers.ts
+
+План змін:
+
+1. Виділити окремі кроки в приватні функції:
+
+- normalizeAndValidateTagNames
+- loadExistingTagsByName
+- createMissingTags
+- resolveTagsForEvent (оркестратор)
```
[the remark this answers is not in the document]
Comment (glckfndr) [reply docs/refactor-roadmap-feedback-2026-03-30.md]: Fixed
